SQL.txt

```diff
@@ -0,0 +1,2686 @@
++-------------+------+---------+------------+------+
+|        2008 | 3001 |   18.69 | 1990-10-03 | 1007 |
+|        2008 | 3006 | 1098.16 | 1990-10-03 | 1007 |
+|        2007 | 3002 |  1900.1 | 1990-10-03 | 1004 |
+|        2006 | 3008 |    4723 | 1990-10-05 | 1001 |
+|        2006 | 3011 | 9891.88 | 1990-10-06 | 1001 |
+|        2004 | 3007 |   75.75 | 1990-10-04 | 1002 |
+|        2004 | 3010 | 1309.95 | 1990-10-06 | 1002 |
+|        2003 | 3005 | 5160.45 | 1990-10-03 | 1002 |
+|        2002 | 3009 | 1713.23 | 1990-10-04 | 1003 |
+|        2001 | 3003 |  767.19 | 1990-10-03 | 1001 |
++-------------+------+---------+------------+------+
+
+5) Find which salespeople currently have orders in the order table. 
+Ans- 
+SELECT 
+  sp.snum, 
+  sp.sname, 
+  o.onum, 
+  o.odate, 
+  o.amt 
+FROM 
+  salespeople sp, 
+  orders o 
+WHERE 
+  sp.snum = o.snum 
+ORDER BY 
+  sp.snum;
+
++------+---------+------+------------+---------+
+| snum | sname   | onum | odate      | amt     |
++------+---------+------+------------+---------+
+| 1001 | Peel    | 3003 | 1990-10-03 |  767.19 |
+| 1001 | Peel    | 3008 | 1990-10-05 |    4723 |
+| 1001 | Peel    | 3011 | 1990-10-06 | 9891.88 |
+| 1002 | Serres  | 3005 | 1990-10-03 | 5160.45 |
+| 1002 | Serres  | 3007 | 1990-10-04 |   75.75 |
+| 1002 | Serres  | 3010 | 1990-10-06 | 1309.95 |
+| 1003 | AxelRod | 3009 | 1990-10-04 | 1713.23 |
+| 1004 | Motika  | 3002 | 1990-10-03 |  1900.1 |
+| 1007 | Rifkin  | 3001 | 1990-10-03 |   18.69 |
+| 1007 | Rifkin  | 3006 | 1990-10-03 | 1098.16 |
++------+---------+------+------------+---------+
+
+6) List names of all customers matched with the salespeople serving them. 
+Ans- 
+SELECT 
+  c.cnum, 
+  c.cname, 
+  s.sname 
+FROM 
+  salespeople s, 
+  customers c 
+WHERE 
+  s.snum = c.snum;
+
++------+----------+---------+
+| cnum | cname    | sname   |
++------+----------+---------+
+| 2001 | Hoffman  | Peel    |
+| 2002 | Giovanni | AxelRod |
+| 2003 | liu      | Serres  |
+| 2004 | Grass    | Serres  |
+| 2008 | Cisneros | Rifkin  |
+| 2006 | Cisneros | Peel    |
+| 2007 | Pereira  | Motika  |
++------+----------+---------+
+
+7) Find the names and numbers of all salespeople who have more than one customer. 
+Ans- 
+SELECT 
+  s.sname, 
+  s.snum, 
+  COUNT(c.snum) AS no_orders 
+FROM 
+  salespeople s, 
+  customers c 
+WHERE 
+  s.snum = c.snum 
+GROUP BY 
+  s.snum, 
+  s.sname 
+HAVING 
+  COUNT(c.snum)> 1;
+
++--------+------+-----------+
+| sname  | snum | no_orders |
++--------+------+-----------+
+| Peel   | 1001 |         2 |
+| Serres | 1002 |         2 |
++--------+------+-----------+
+
+8) Count the orders of each of the salespeople and output the results in descending order. 
+Ans- 
+SELECT 
+  s.sname, 
+  s.snum, 
+  COUNT(c.snum) AS no_orders 
+FROM 
+  salespeople s, 
+  customers c 
+WHERE 
+  s.snum = c.snum 
+GROUP BY 
+  s.snum, 
+  s.sname 
+ORDER BY 
+  COUNT(c.snum) DESC;
+
++---------+------+-----------+
```

Comment:
Incorrect output.

Reply:
I have have made all required changes ma'am.

Thank you